src/teb_local_planner_ros.cpp

```diff
@@ -869,13 +869,28 @@ double TebLocalPlannerROS::estimateLocalGoalOrientation(const std::vector<geomet
 }
       
       
-void TebLocalPlannerROS::saturateVelocity(double& vx, double& vy, double& omega, double max_vel_x, double max_vel_y, double max_vel_theta, double max_vel_x_backwards) const
+void TebLocalPlannerROS::saturateVelocity(double& vx, double& vy, double& omega, double max_vel_x, double max_vel_y, double max_vel_theta, double max_vel_x_backwards, double min_vel_x) const
 {
-  double ratio_x = 1, ratio_omega = 1, ratio_y = 1;
-  // Limit translational velocity for forward driving
-  if (vx > max_vel_x)
-    ratio_x = max_vel_x / vx;
-  
+  double ratio_x = 1, ratio_omega = 1, ratio_y = 1, ratio_x_min = 1;
+
+  // backward driving
+  if(vx < 0)
+  {  
+    if (vx < -max_vel_x_backwards)
+      ratio_x = - max_vel_x_backwards / vx;
+    else if (vx > -min_vel_x)
+      ratio_x = - min_vel_x / vx;
+  }
+  else // forward driving
+  {
+    if (vx > max_vel_x)
+      ratio_x = max_vel_x / vx;
+    else if (vx < min_vel_x)
+    {
+      ratio_x =   min_vel_x / vx;
+    }
```

Comment:
please adhere to current code style 
```suggestion
      ratio_x = min_vel_x / vx;
```